### PR TITLE
[webgpu] Tweak WGSL entry point emission so '{' isn't visually swallowed

### PR DESCRIPTION
**Pull request by hujiajie**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6764)

### CODE REVIEW

**Review by gyagp — APPROVED** [review]
LGTM. Thanks for the unification!

**Review by tellink1 — APPROVED** [review]

File: tfjs-backend-webgpu/src/webgpu_program.ts [hunk 1]
```diff
@@ -103,23 +103,45 @@ export function getCoordsXYZ(index: number): string {
...
+  switch (params.length) {
+    case 0:
+      snippet = `
+        ${getWorkGroupSizeString()}
+        fn _start(@builtin(local_invocation_id) LocalId : vec3<u32>,
+                  @builtin(global_invocation_id) GlobalId : vec3<u32>,
+                  @builtin(num_workgroups) NumWorkgroups : vec3<u32>) {
+          localId = LocalId;
+          globalId = GlobalId;
+          numWorkgroups = NumWorkgroups;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/webgpu_program.ts:119]: Can you check if `numWorkgroups = NumWorkgroups;` is still needed for this case? I expect if we use non-linear work group size, we can totally remove the dependency on `num_workgroups` and `getGlobalIndex()`. I am ok to do it in a follow-up PR. Thanks.

**Review by qjia7 — APPROVED** [review]
LGTM with a comment. But I am ok to merge this PR first and see if we can do further cleanup to address the comment in follow-up PRs.

File: tfjs-backend-webgpu/src/webgpu_program.ts (hunk 1 above)
Comment (hujiajie) [reply tfjs-backend-webgpu/src/webgpu_program.ts:119]: I don't feel it's safe to assume `getGlobalIndex()` is unused elsewhere. E.g. we currently have this dependency chain in depth-wise conv2d: `main() -> getOutputCoords() -> getGlobalIndex()`.

Comment (qjia7) [reply tfjs-backend-webgpu/src/webgpu_program.ts:119]: `getGlobalIndex()` is only called when `x.length === outRank` in `getOutputCoords`, which means a flat dispatch layout. Let's keep it for now. Maybe it's not necessary to worry about the very small overhead of  `NumWorkgroups `.

Comment (hujiajie) [reply tfjs-backend-webgpu/src/webgpu_program.ts:119]: I do find the `x.length === outRank` branch is taken while running depthwise_conv2d_test, is this expected?

**Pull request merged**